objects/Editors/contained/PowerEditor/script.lua

```diff
@@ -487,6 +487,47 @@ function populateThreshold()
     end
 end
 
+function updateReminder(player)
+    if currentCard == nil then
+        return
+    end
+    local hits = upCast(currentCard, 0.4, 0, {"Generic"}, "Reminder")
+    local location = nil
+    for _, entry in pairs(hits) do
+        location = entry.call("getImageLocation", {obj = currentCard})
+        entry.destroy()
+    end
+
+    local state = {}
+    if currentCard.script_state ~= "" then
+        state = JSON.decode(currentCard.script_state)
+    end
+    state.reminder = location
+    currentCard.script_state = JSON.encode(state)
+
+    local opString = "Updated"
+    if location == nil then
+        opString = "Reset"
+    end
+    player.broadcast(opString .. " reminder image location for " .. currentCard.getName() .. ".", Color.SoftBlue)
+end
+function populateReminder()
+    if currentCard == nil then
+        return
+    end
+    local location = Global.call("getReminderLocation", {obj = currentCard})
```

Comment:
should we add a return here if `location` is null? e.g. nothing to populate if it hasn't been set before

Reply:
`location` *can't* actually be `nil`: `getReminderLocation()` only returns `nil` if called with `obj = nil` (which is ruled out by the return earlier in this function) or if called on an `obj` that is not a card or spirit, and `currentCard`, if not `nil`, is always a card. This is 'cause `getReminderLocation()` returns the default location if called on a card or spirit without a location set.

That said, guess there's no harm tossing a `nil` check *just in case*.

Reply:
ah gotcha, hadn't gotten around yet to the global script changes when i made this comment